### ISSUE TITLE
/etc/php.ini is not generated when osl-php::packages is included instead of osl-php::default

### ISSUE THREAD

**Issue by TravisWhitehead** [osuosl-cookbooks/osl-php#10]
**TL;DR** Should we just include `php::ini` in `osl-php::packages`, as a quick fix, or address (what seems to me) like a larger design problem?

**Background Information:**
The `php::ini recipe` generates `/etc/php.ini` with the options in `node['php']['directives']`, and is included in `php::default`.

It appears that various recipes in our cookbooks will install PHP in one of two ways:

1. `include_recipe 'osl-php::packages'`

    All this does is install all of the packages specified in `node['osl-php']['packages']`, and `php::ini` is never included.

2. `include_recipe 'osl-php'`

    This includes `osl-php::packages` and then `php::default` (which includes `php::ini`). `php::default` just includes `php::package`, which does the same thing as our `osl-php::packages`, except it uses the `node['php']['packages']` attribute.

Note that the second method will fail if `node['php']['packages']` is not changed/overriden, as our specified packages will conflict with what the php cookbook specifies. The `osl-php` cookbook does not override it. The only node converging `osl-php::default` is `web2.osuosl.org`, and the `node['php']['packages']` is cleared in `role[web2]`.

**My question:**
Knowing that the default recipe fails without overriding the `node['php']['packages']` attribute, and that `php::default` does essentially what `osl-php::packages` does (but with `php::ini` included, which we want)... Doesn't it make more sense just to do away with `osl-php::packages`, use `osl-php::default` as the recipe to install PHP, and set `node['php']['packages'] = node['osl-php']['packages']` in that default recipe?

Of course, this would require altering all the cookbooks including `osl-php::packages`, which we might not want to do... But just including `php::ini` feels like skirting around a larger problem. If an attribute needs to be changed for a recipe not to fail, shouldn't that be done in the recipe?

Thoughts?

**Links for Reference:**
* [php::default](https://github.com/chef-cookbooks/php/blob/master/recipes/default.rb)
* [php::package](https://github.com/chef-cookbooks/php/blob/master/recipes/package.rb)
* [osl-php::default](https://github.com/osuosl-cookbooks/osl-php/blob/master/recipes/default.rb)
* [osl-php::packages](https://github.com/osuosl-cookbooks/osl-php/blob/master/recipes/packages.rb)

**Comment by TravisWhitehead** [comment]
@knightsamar @fahlmant 

Please let me know what you think about the best way to fix this. Including php::ini in osl-php::packages is an easy quick-fix for getting proj-phpbb cookbook to work in the meantime, but I don't think it's a good solution in the long term.

**Comment by ramereth** [comment]
This cookbook has always had design problems unfortunately since it's inception. Also make sure you're trying against the version of the cookbook we use in production (we probably should add a pessimistic version lock in metadata.rb if we haven't already).

Would simply adding ``php::ini`` in ``osl-php::packages`` solve the problem at hand at least in the short term? What implications would it have?

**Comment by TravisWhitehead** [comment]
Yep, adding `include_recipe 'php::ini'` to `osl-php::packages` does solve the problem. I think that's an appropriate fix for the sake of getting `/etc/php.ini` managed by Chef for the nodes that only include `osl-php::packages`, in the meantime.

This change affects all nodes that converge `osl-php::packages` and not `osl-php::default` (as default already does `php::ini` through `php::default`).

`knife search node -i 'recipes:osl-php\:\:packages AND NOT recipes:osl-php\:\:default'` outputs:

```
crowd1.phpbb.com
bamboo1.phpbb.com
scripts.phpbb.com
secure1.phpbb.osuosl.org
www4.phpbb.com
demo1.phpbb.osuosl.org
deluge.osuosl.org
www2.phpbb.com
www3.phpbb.com
```
I've already looked into the consequences of this change for all of the affected nodes from the phpbb environment, so the only one to worry about is `deluge.osuosl.org`. (All of the phpbb nodes but bamboo1 will remain the same, and I've brought that one up in ticket 29428).

I'll look into the consequences this has for `deluge.osuosl.org`.

**Comment by TravisWhitehead** [comment]
`deluge.osuosl.org` 's `php.ini` would remain exactly the same after this change. The values specified in the `proj-deluge::php` recipe match production, as does the upstream template itself. (Although the suite's tests don't match the values in its recipe/production, so that's something we should probably fix.)

Knowing this, it sounds like including `php::ini` in `osl-php::packages` is what we should do, in the meantime. I'll wait on opening a PR until I hear back from the phpBB folks in ticket 29428 (so we don't screw up `bamboo1.phpbb.com`).

**Comment by TravisWhitehead** [comment]
#13 (more or less) addresses my complaints.
